### PR TITLE
Add YouTube widget

### PR DESCRIPTION
**Pull request by mechakin**
Implements the youtube widget as seen in #270

Currently able to play videos if sent a youtube video url, a share link (a youtu.be link), or a playlist link

### CODE REVIEW

**Comment by Melkeydev** [comment]
Going to be looking at this PR tonight

File: src/components/Youtube/YoutubeVideo.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  const [inputText, setInputText] = useState("");
+  const [videoId, setVideoId] = useState(defaultVideoId);
+  const regex = new RegExp("[^=|\/]*$");
+
+  const handleVideoChange = (streamName: string) => {
+    const id = streamName.match(regex)[0];
+    
+    if (id.length > 0) setVideoId(id);
+
+    if (id.length > 11) setVideoId(playlistText + id);
```
Comment (Melkeydev) [inline src/components/Youtube/YoutubeVideo.tsx:19]: Why hard coded to 11?

Comment (Melkeydev) [inline src/components/Youtube/YoutubeVideo.tsx:19]: I am not sure about these two lines here

**Review by Melkeydev — COMMENTED** [review]
Added minor comments

**Comment by Melkeydev** [comment]
Also - there are some conflicts

File: src/components/Youtube/YoutubeVideo.tsx (hunk 1 above)
Comment (mechakin) [reply src/components/Youtube/YoutubeVideo.tsx]: The regex above makes sure that it gets the youtube video/playlist id from the link the user provides, for an individual video it is guaranteed to be 11 characters long which is why it's hardcoded that way. However, there was an issue with it where it wouldn't properly work with videos that have a timestamp attached (ex. https://www.youtube.com/watch?v=aonx9JRry7o**&t=7s**). The new commits should fix this issue

**Comment by mechakin** [comment]
Implemented it so there weren't any checks for id to be a specific length anymore

**Comment by Melkeydev** [comment]
@mechakin are you still interested in merging this? I am prioritising Astrostation - all this needs it to fix merge conflicts

**Comment by mechakin** [comment]
yeah i am interested in merging this, just fixed the merge conflicts

**Pull request merged**